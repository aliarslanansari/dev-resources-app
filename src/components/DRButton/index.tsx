import { Button, ButtonProps, CircularProgress } from '@material-ui/core'
import { memo } from 'react'

interface DRButtonPropsTypes extends ButtonProps {
  onClick?: () => void
  loading?: boolean
  label: string
}

function DRButton(props: DRButtonPropsTypes) {
  const { onClick, loading, label } = props
  return (
    <Button
      data-testid='button'
      variant='outlined'
      color='secondary'
      onClick={onClick}
      disabled={loading}
      {...props}
    >
      {loading && <CircularProgress color='secondary' size={'2rem'} />}
      {!loading && label}
    </Button>
  )
}

export default memo(DRButton)
