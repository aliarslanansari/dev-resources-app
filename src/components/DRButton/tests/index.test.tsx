import { render } from '@testing-library/react'
import DRButton from '../index'

describe('<DRButton />', () => {
  it('should render and match the snapshot', () => {
    const onClick = () => {
      console.log('test')
    }
    const { baseElement } = render(
      <DRButton label='Create Post' style={{ margin: 16 }} onClick={onClick} />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 Button component', () => {
    const { getAllByTestId } = render(
      <DRButton label='Create Post' style={{ margin: 16 }} />
    )
    expect(getAllByTestId('button').length).toBe(1)
  })
})
