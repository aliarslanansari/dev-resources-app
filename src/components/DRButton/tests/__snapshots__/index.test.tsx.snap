// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DRButton /> should render and match the snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
      data-testid="button"
      label="Create Post"
      style="margin: 16px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Create Post
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
